src/main/scala/com/gu/identity/play/AuthenticatedIdUser.scala

```diff
@@ -81,14 +81,16 @@ object AccessCredentials {
       val collectionSigner = new CollectionSigner(new StringSigner(new DsaService(identityKeys.publicDsaKey, null)), LiftJsonConfig.formats)
 
       // Adapted from https://github.com/guardian/identity/blob/8663b03/identity-api-client-lib/src/main/java/com/gu/identity/client/IdentityApiClient.java#L321-L334
-      def extractUserDataFromToken(tokenString: String) = {
-        val cryptoAccessToken = collectionSigner.getValueForSignedStringJava(tokenString, classOf[CryptoAccessToken])
-
-        if (cryptoAccessToken.expiryTime < DateTime.now) {
-          Left(s"Token: $tokenString has expired")
-        } else if (cryptoAccessToken.targetClient != targetClientId) {
-          Left(s"Token: $tokenString was not targeted for the client '$targetClientId'")
-        } else Right(cryptoAccessToken.getUser)
+      def extractUserDataFromToken(tokenString: String): Either[String, User] with Product with Serializable = {
```

Comment:
Ok, this can be just `Either[String, User]` (drop the `with Product with Serializable`!) and then :+1: